src/timessquare/handlers/v1/handlers.py

```diff
@@ -10,58 +12,76 @@
     context_dependency,
 )
 
-from .models import Index, Page, PostPageRequest
+from .models import Index, Page, PageSummary, PostPageRequest
 
 __all__ = ["v1_router"]
 
-v1_router = APIRouter()
+v1_router = APIRouter(tags=["v1"])
 """FastAPI router for all external handlers."""
 
 
 @v1_router.get(
     "/",
-    description=(
-        "Metadata about the v1 REST API, including links to "
-        "documentation and endpoints."
-    ),
     response_model=Index,
     summary="V1 API metadata",
 )
 async def get_index(
     request: Request,
 ) -> Index:
+    """Get metadata about the v1 REST API, including links to documentation and
+    endpoints.
+    """
     metadata = get_metadata(
         package_name="times-square",
         application_name=config.name,
     )
-    return Index(metadata=metadata, api_docs=f"{request.url}docs")
+    doc_url = request.url.replace(path=f"/{config.name}/docs")
+    return Index(metadata=metadata, api_docs=str(doc_url))
 
 
 @v1_router.get(
     "/pages/{page}",
-    description=(
-        "Get metadata about a page resource, which models a webpage that is "
-        "rendered from a parameterized Jupyter Notebook."
-    ),
     response_model=Page,
     summary="Page metadata.",
     name="get_page",
 )
 async def get_page(
     page: str, context: RequestContext = Depends(context_dependency)
 ) -> Page:
+    """Get metadata about a page resource, which models a webpage that is
+    rendered from a parameterized Jupyter Notebook.
+    """
```

Comment:
This is just a personal preference thing so this isn't an objection, but I prefer keeping this in the description field rather than using a docstring. A lot of that is that the variance from the PEP 257 requirement that the first line of the docstring truly be a single line, not wrapped, irrationally bugs me. Also, since I never write a proper docstring for handler functions (I never write the parameters section, for instance, nor do I think that's useful to do), it seemed cleaner to not use it at all. Just an FYI for why my code tends to use description instead.

Reply:
It's a lot easier to write and maintain extended descriptions via a docstring and since we're optimizing for the Open API doc experience here rather than Python API docs (which I probably wouldn't generate from handler modules anyways), I think it's pragmatic to ignore PEP 257 and embrace the FastAPI functionality.